index.bs

```diff
@@ -190,19 +187,45 @@ Compute the layout shift {#sec-compute-layout-shift}
+    1. Set <var>layoutShift</var> to the <var>impactedFraction</var> multiplied by the <var>maxShiftFraction</var>.
     1. If <var>layoutShift</var> is greater than 0:
         1. Create a new {{LayoutShift}} object |newEntry|.
         1. Set |newEntry|'s {{PerformanceEntry/name}} attribute to <code>"layout-shift"</code>.
-        1. Set |newEntry|'s {{PerformanceEntry/entryType}} attribute to <code>"layoutShift"</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/entryType}} attribute to <code>"layout-shift"</code>.
         1. Set |newEntry|'s {{PerformanceEntry/startTime}} attribute to <a>current high resolution time</a>.
         1. Set |newEntry|'s {{PerformanceEntry/duration}} attribute to 0.
-        1. Set |newEntry|'s {{LayoutShift/value}} attribute to <var>layoutShift</var>.
+        1. Set |newEntry|'s <dfn attribute for=LayoutShift>value</dfn> attribute to <var>layoutShift</var>.
+        1. Set |newEntry|'s <dfn attribute for=LayoutShift>lastInputTime</dfn> attribute
+            to the time of the most recent <a>excluding input</a>, or 0 if no excluding input has occurred
+            during the browsing session.
+        1. Set |newEntry|'s <dfn attribute for=LayoutShift>hadRecentInput</dfn> attribute
+            to <code>true</code> if {{LayoutShift/lastInputTime}} is less than 500 milliseconds in the past,
+            and <code>false</code> otherwise.
         1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a> |newEntry| object.
 </div>
 
 NOTE: This computation ensures that the layout shift takes into account both the fraction of the viewport that has been impacted by layout stability as well as the greatest impact to any given element in the viewport.  This is to recognize that a large element that moves only a small distance can have a small impact on the perceived stability of the page.
 
+Input Exclusion {#sec-input-exclusion}
+--------------------------------------
+
+An <dfn export>excluding input</dfn> is any event from an input device which signals a user's
+active interaction with the page, or any event which directly changes the size of the <a>viewport</a>.
+
+Excluding inputs generally include <a href="https://www.w3.org/TR/uievents/#event-type-mousedown">mousedown</a>,
+<a href="https://www.w3.org/TR/uievents/#keydown">keydown</a>, and
+<a href="https://www.w3.org/TR/pointerevents/#the-pointerdown-event">pointerdown</a>.
+However, an event whose only effect is to begin or update a fling or scroll gesture
+is not an excluding input.
+
+The user agent may delay the reporting of layout shifts after a
+<a href="https://www.w3.org/TR/pointerevents/#the-pointerdown-event">pointerdown</a> event
+until such time as it is known that the event does not begin a fling or scroll gesture.
+
+The <a href="https://www.w3.org/TR/uievents/#event-type-mousemove">mousemove</a> and
```

Comment:
Ideally the timestamp / boolean would be set/unset somewhere. Is there an issue to improve this?

Reply:
Sorry, I don't understand what you mean here.  Can you elaborate?

Reply:
Oh I realize I commented in a wrong spot. I meant the |lastInputTime| and |hadRecentInput| seem to come out nowhere. I think in this case it's fine as it seems clear enough to me how they are meant to be computed.

Reply:
No inferences are necessary as to how these are computed - the explicit instructions to set them appear in "Compute the layout shift", along with the other attributes of the LayoutShift object.